src/Butil/Bit.Butil/Scripts/userAgent.ts

```
@@ -0,0 +1,595 @@
+        if (/\bSimulator\b/i.test(ua)) {
+            product = (product ? product + ' ' : '') + 'Simulator';
+        }
+
+        // Detect Opera Mini 8+ running in Turbo/Uncompressed mode on iOS.
+        if (name == 'Opera Mini' && /\bOPiOS\b/.test(ua)) {
+            description.push('running in Turbo/Uncompressed mode');
+        }
+
+        // Detect iOS.
+        else if (/^iP/.test(product)) {
+            name || (name = 'Safari');
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)